test_unstructured/chunking/test_title.py

```diff
@@ -91,12 +99,12 @@ def test_chunk_by_title():
 
     assert chunks[0].metadata == ElementMetadata(emphasized_text_contents=["Day", "day"])
     assert chunks[3].metadata == ElementMetadata(
-        regex_metadata=[{"text": "A", "start": 11, "end": 12}],
+        regex_metadata={"a": [RegexMetadata(text="A", start=11, end=12)]}
```

Comment:
why did the data structure of the value change here?

would this get json serialized / deserialized correctly?

Reply:
The type of `ElementMetadata.regex_metadata` is `Dict[str, List[RegexMetadata]]`. `RegexMetadata` is a `TypedDict` like `{"text": "this matched", "start": 7, "end": 19}`.

My assumption is you can specify multiple regexes, each with a name like "mail-stop", "version", etc., and each of those may produce its own set of matches, like:

```python
>>> element.regex_metadata
{
    "mail-stop": [{"text": "MS-107", "start": 18, "end": 24}],
    "version": [
        {"text": "current: v1.7.2", "start": 7, "end": 21},
        {"text": "supersedes: v1.7.2", "start": 22, "end": 40},
    ],
}
```

My guess is that this "multiple-regexes" behavior was originally "single-regex" and when it was changed the tests and chunking code wasn't updated. But that's just my guess. Not sure who might remember about that.

Reply:
On the serializing/deserializing question, I don't see why it wouldn't work fine, these are all primitive data structures and JSON serializable, but let me take a closer look.

Reply:
Okay, looking at the serde code, it looks to me like this should round-trip fine. However, there are no tests confirming this that I can find, at least not in `test_unstructured/documents/test_elements.py`. If you want I can ticket this and write a more thorough test. I'll whip up an informal one here just to satisfy myself it actually works, but I'm thinking that's probably outside the scope of this PR to actually add it in, do you agree?

Note that the changes in this PR don't affect the type of `ElementMetadata.regex_metadata` or its serde in any way, they just turned up the type error when I cleaned up typing in this neighborhood and that's what led to these two changes.

They _would_ however (indirectly) affect the serde of the chunks (`CompositeElement` objects mostly) in that regex meta would have been stripped from second and later section elements, so the `CompositeElement` would only contain matches present in the first element.

Reply:
Yep, round-trips fine in my test.

Reply:
I added a test to confirm JSON round-trip of `ElementMetadata.regex_metadata`.

Reply:
> But this account is just my guess. Not sure who might remember about that.

git blame can provide that info :)

Reply:
@cragwolfe okay, I did the forensics, turns out it was just a mistake from the start. Good advertisement for strict data typing :)

* 06/16/2023 4ea71683 - regex_metadata feature was added by Matt Robinson in substantially its current form: (`Dict[str, List[RegexMetadata]]`).
* 08/29/2023 f6a745a7 - chunk by titles was added by Matt with the wrong type for regex_metadata in the tests, as we found it.